File: backend/EasyCourse.Core/Entities/Section.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace EasyCourse.Core.Entities;

public class Section
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid SectionId { get; set; }
    public Guid CourseId { get; set; }
    public int Order { get; set; }
    public string Title { get; set; } = string.Empty;
    public int ReadingTime { get; set; } = 0;

    public string SectionData { get; set; } = string.Empty;
    public string SectionQuestions { get; set; } = string.Empty;
    [JsonIgnore]
    public Course? Course { get; set; } // navigation property
    public DateTime? LastUpdated { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/section.ts
export interface Section {
  sectionId?: string;
  title?: string;
  order?: number;
  courseId?: string;
  sectionData?: any;
  sectionQuestions?: any;
  readingTime?: number;
  lastUpdated?: Date;
}

export type QuestionType = "multiple-choice" | "true-false" | "short-answer";

export interface QuestionOption {
  id: string;
  text: string;
  isCorrect?: boolean;
}

export interface Question {
  id: string;
  type: QuestionType;
  questionText: string;
  options?: QuestionOption[];
  correctAnswer?: string;     
}